src/widgets/stats_editor.rs

```diff
@@ -100,12 +105,17 @@ impl<S: Stats> Widget for StatsEditor<S> {
             for datum in data {
                 match datum {
                     Datum::Int { label, value, min, max } => {
-                        if ui.input_int(label, value).build() {
+                        if max >= 0 && ui.input_int(label, value).build() {
```

Comment:
What's the purpose of this check? We shouldn't make assumptions on enforcing positive values, otherwise we'd be better off using unsigned types instead.

Also, I don't think this is the appropriate place for a check like this. If the user mistakenly puts a negative `max` value, they won't be able to understand where their error comes from as the widget will fail silently as a consequence of the check.

`Datum` can only be instantiated via the explicit constructors above, so if anything we should move this check in there and make them either return `Result` (verbose but explicit) or just swap `min` and `max` (implicit but potentially confusing), but honestly I think we should just get rid of it altogether. Same for the other two similar checks below.

Reply:
This is used to hide blessing values' editing in versions older than DLC for Elden Ring